packages/client/src/interface/navigation/channels/ServerSidebar.tsx

```diff
@@ -370,26 +370,25 @@ function Entry(
               <MdPersonAdd {...iconSize("14px")} />
             </a>
 
-            <a
-              use:floating={{
-                tooltip: {
-                  placement: "top",
-                  content: "Edit Channel",
-                },
-              }}
-              onClick={(e) => {
-                e.preventDefault();
-                openModal({
-                  type: "settings",
-                  config: "channel",
-                  context: props.channel,
-                });
-              }}
-            >
-              <MdSettings {...iconSize("14px")} />
-            </a>
-          </>
-        }
+    <Show when={props.channel.server?.havePermission("ManageChannel")}>
```

Comment:
There are quite a few permissions that could allow access to channel settings, namely managing permissions or webhooks. (https://developers.revolt.chat/developers/api/permissions.html)